src/protocol/ipa/mod.rs

```diff
@@ -155,24 +156,60 @@ impl<F: Field + Sized, T: Arithmetic<F>> Resharable<F> for IPAModulusConvertedIn
     }
 }
 
+/// Describes a request to execute IPA protocol. Consists of inputs for the computation
+/// and query parameters
+pub struct Query<'a, F: Field, MK: BitArray = MatchKey, BK: BitArray = BreakdownKey> {
+    input: &'a [IPAInputRow<F, MK, BK>],
+    /// Maximum contribution allowed per match key/user.
+    per_user_credit_cap: u32,
+    /// ?
+    max_breakdown_key: u128,
+    /// That many bits will be sorted at the same time. Setting it to 1 would mean IPA sorts
+    /// 1 bit at a time.
```

Comment:
Experimentally, we have determined that 3 is the optimal number to reduce communication overhead.

Reply:
It's fine to have this as a tuning parameter, but please set a default if you annotate this.